### PR TITLE
Fix exports/1 in IEx for long fun names

### LINKED ISSUE #11968: IEx.Helpers.exports/1 not breaking line
### Elixir and Erlang/OTP versions

Erlang/OTP 24 [erts-12.3.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1]

Elixir 1.13.4 (compiled with Erlang/OTP 24)

### Operating system

macOS 13 (build 22A5266r)

### Current behavior

IEx `exports/1` helper function does not seem to be printing the table correctly in some cases.

```
iex(5)> exports Calendar.UTCOnlyTimeZoneDatabase
time_zone_period_from_utc_iso_days/2time_zone_periods_from_wall_datetime/2
```

https://github.com/elixir-lang/elixir/blob/8640def5c85a90e070f6b1f4066568b377cdb972/lib/iex/lib/iex/helpers.ex#L795

### Expected behavior

I was expecting a line break or a space, for example:

```diff
- time_zone_period_from_utc_iso_days/2time_zone_periods_from_wall_datetime/2
+ time_zone_period_from_utc_iso_days/2 time_zone_periods_from_wall_datetime/2

# or

- time_zone_period_from_utc_iso_days/2time_zone_periods_from_wall_datetime/2
+ time_zone_period_from_utc_iso_days/2
+ time_zone_periods_from_wall_datetime/2
```

### PR DESCRIPTION
**Pull request by sabiwara**
Close #11968

The problem was that `print_table` uses a [max_length](https://github.com/elixir-lang/elixir/blob/main/lib/iex/lib/iex/helpers.ex#L790-L791=) in which case `String.pad_trailing` would not be adding any space.

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you @sabiwara! I wonder if maybe we should fix this on `print_table` instead?

We can make `print_table` pass `width - 1` and then always add a space or a newline after accordingly. The default function continues to be `String.pad_trailing/2`. WDYT?

**Comment by sabiwara** [comment]
> I wonder if maybe we should fix this on `print_table` instead?

You're right, I confirmed that `ls` which is also relying on `print_table` has the same issue. Will fix it in `print_table` 👍

File: lib/iex/lib/iex/helpers.ex [hunk 1]
```diff
@@ -802,8 +802,11 @@ defmodule IEx.Helpers do
           length
         end
 
-      IO.write(printer.(item, offset))
-      length + offset
+      printed = printer.(item, offset)
+      actual_offset = String.length(printed) + 1
```
Comment (sabiwara) [inline lib/iex/lib/iex/helpers.ex:806]: By recomputing the `offset` whenever the length is exceeding the expected target, we avoid this kind of issue:

```
iex(3)> ls
foobauhoiuhfsdiohfoihfajojfoiajfojasfojoicjoicjocjcoodfosdosejoe fseventsd-uuid                     powerlog
foo.bar
```

and get the following instead:

```
iex(2)> ls
foobauhoiuhfsdiohfoihfajojfoiajfojasfojoicjoicjocjcoodfosdosejoe fseventsd-uuid
powerlog                           foo.bar
```

**Comment by sabiwara** [comment]
@josevalim is [e397293](https://github.com/elixir-lang/elixir/pull/11972/commits/e3972938fa933a3e95395018098c09923e03ec18) what you had in mind?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**